jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/file/AssetFileGenerator.java

```diff
@@ -211,12 +213,16 @@ public void generateFiles(List<SubscriptionVH> subscriptions, String filePath) {
 					.fetchPolicyDefinitionDetails(subscription);
 			ExecutorService executor = Executors.newCachedThreadPool();
 
+			List<String> targetTypeList = new CopyOnWriteArrayList<>();
```

Comment:
can we make `hangingTergetTypes` and then add additional list like `failingTargetTypes` and then add types into that  list if file generation failed (in catch section)? does it make sense?

Reply:
hangingTargetTypes is okay but we can't be accurate with failingTargetTypes because many exceptions are caught inside individual collectors. Sometimes, some exceptions occurring in collectors are ignored and data collection continues. In that case, logging failingTargetTypes will not give a clear picture on what are genuine/important exceptions.

Reply:
got it, let's keep it as it is now then